subsys/net/lib/fota_download/src/fota_download.c

```diff
@@ -243,7 +243,11 @@ static int download_client_callback(const struct download_client_evt *event)
 					"used by dfu_target.");
 			}
 			first_fragment = true;
-			send_error_evt(FOTA_DOWNLOAD_ERROR_CAUSE_DOWNLOAD_FAILED);
+			if (event->error == -ECONNRESET || event->error == -ETIMEDOUT) {
+				send_error_evt(FOTA_DOWNLOAD_ERROR_CAUSE_CONNECTION_LOST);
```

Comment:
Should we make here the distinction that TIMEOUT is a CONNECTION_LOST and the if peer is actively closing the connection (ECONNRESET) then it is DOWNLOAD_FAILED?